src/forms/signpost/SignPost.tsx

```diff
@@ -0,0 +1,152 @@
+        window.removeEventListener("click", this.handleDocumentClick as any, true);
+    };
+
+    handleDocumentClick = (evt: React.MouseEvent<HTMLElement>) => {
+        if (!this.state.isOpen) return;
+        const target = (evt.target as any) as HTMLElement;
+
+        const el = this.refChild.current;
+        if (!el || typeof el === "string") {
+            console.warn("wrong element type");
+            return;
+        }
+        if (!el.contains(target)) {
+            this.toggle(false);
+        }
+    };
+
+    componentDidUpdate() {
+        const {isOpen} = this.state;
+        if (!isOpen) return;
+        const childWidth = (this.refChild.current as HTMLDivElement).clientWidth;
+        const childHeight = (this.refChild.current as HTMLDivElement).clientHeight;
+        const parentWidth = (this.refParent.current as HTMLDivElement).clientWidth;
+        const parentHeight = (this.refParent.current as HTMLDivElement).clientHeight;
+
+        const {direction} = this.props;
+        let transformVal =
+            direction && calculateAxisPosition(parentHeight, parentWidth, childHeight, childWidth, direction);
+        this.setState({transformVal: transformVal});
+    }
+
+    render() {
+        const {isOpen, transformVal} = this.state;
+        const {direction, style, children} = this.props;
+        let setDirection = direction ? direction : "top-left";
+        return (
+            <div ref={this.refParent} className="signpost" style={{position: "relative"}}>
+                <Button
+                    className={classNames([
+                        "signpost-action", //prettier
+                        "btn-small",
+                        "btn-link",
+                        "signpost-trigger",
+                        isOpen && "active",
+                    ])}
+                    onClick={this.handleButtonClick.bind(this)}
+                >
+                    <Icon shape="info" size={26} />
+                </Button>
+                {isOpen && (
+                    <div
+                        ref={this.refChild}
+                        className={classNames([
+                            "signpost-content",
+                            setDirection, //prettier
+                        ])}
+                        style={{
+                            ...style,
+                            position: "absolute",
+                            top: "0px",
+                            bottom: "auto",
+                            left: "0px",
+                            right: "auto",
+                            transform: transformVal,
+                        }}
+                    >
+                        <div className="signpost-flex-wrap">
+                            <div className="popover-pointer" />
+                            <div className="signpost-content-header">
+                                <Button className="signpost-action close" onClick={this.handleButtonClick.bind(this)}>
```

Comment:
Better to create an arrow function that will be bound to "this" of the component. Here you will bind this function to this at each render call

Reply:
Done